internal/pathpolicy/path_policy.go

```diff
@@ -24,6 +24,15 @@ func NewPathPolicies(entries map[string]PathPolicy) *PathPolicies {
 	return NewPathTrieFromMap(noType)
 }
 
+func mergePolicyPath(entry_1 map[string]PathPolicy, entry_2 map[string]PathPolicy) map[string]PathPolicy {
+	for k, v := range entry_2 {
+		if _, exists := entry_1[k]; !exists {
+			entry_1[k] = v
+		}
+	}
+	return entry_1
+}
```

Comment:
This doesn't fix the issue.  You're still modifying `entry_1` which is the global `defaultMountpointPolicies` when you're creating the `OstreeMountpointPolicies`.

Run this test, for example, to see the issue:
```go
func TestPathPolicyMerge(t *testing.T) {
	_, ok := defaultMountpointPolicies["/ostree"]
	assert.False(t, ok, "/ostree found in defaultMountpointPolicies")
}
```
You can copy the first map and then merge the second into it to make this work as intended, or make it so that the function explicitly modifies one of the two maps, and use it that way, but that might be trickier to communicate.

Also, an aside I didn't mention earlier: `entry_1` and `entry_2` aren't great variable names here and they're not idiomatic for the language (underscores).

Reply:
Your two versions do raise an interesting question though.  Which policy map wins when both being merged have a policy for the same path?  In the first version it was the second entry and now in the second version it's the first.  This should also be made explicit, probably in the docstring of the function, however it ends up working.

Reply:
This does not apply any more with the new approach